ui/lib/pki/addon/components/page/pki-key-details.ts

```diff
@@ -0,0 +1,42 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import errorMessage from 'vault/utils/error-message';
+interface Args {
+  key: {
+    rollbackAttributes: () => void;
+    destroyRecord: () => void;
+    backend: string;
+    keyName: string;
+    keyId: string;
+  };
+}
+
+export default class PkiKeyDetails extends Component<Args> {
+  @service declare router: { transitionTo: (route: string) => void };
```

Comment:
Would love to see the types here moved to either a typescript declare module, or imported directly from some stub types folder. Cloud-UI also has a pattern of doing `@service declare readonly myService` but I'm not finding any luck with why. Couldn't hurt though!

Reply:
Okay! I wasn't sure how to do the module way. I had noticed the other way for this was: 
```js
 import RouterService from '@ember/routing/router-service';
 service declare readonly router: RouterService;
```

what do you think?